jwst/saturation/saturation.py

```diff
@@ -160,6 +162,31 @@ def irs2_flag_saturation(input_model, ref_model, n_pix_grow_sat):
                                         irs2_mask, detector)
             # check for saturation
             flag_temp = np.where(sci_temp >= sat_thresh, SATURATED, 0)
+            if group == 2:
+                # Identify groups which we wouldn't expect to saturate by the third group,
+                # on the basis of the first group
+                scigp1 = x_irs2.from_irs2(data[ints, 0, :, :], irs2_mask, detector)
+                mask = scigp1 / np.mean(read_pattern[0]) * read_pattern[2][-1] < sat_thresh
+
+                # Identify groups with suspiciously large values in the second group
+                scigp2 = x_irs2.from_irs2(data[ints, 1, :, :], irs2_mask, detector)
+                mask &= scigp2 > sat_thresh / len(read_pattern[1])
+
+                # Identify groups that are saturated in the third group
+                gp3mask = np.where(flag_temp & SATURATED, True, False)
+                mask &= gp3mask
+
+                # Flag the 2nd group for the pixels passing that gauntlet in the 3rd group
+                dq_temp = np.zeros_like(mask,dtype='uint8')
+                dq_temp[mask] = SATURATED
+                # flag any pixels that border saturated pixels
+                if n_pix_grow_sat > 0:
```

Comment:
Is it really worthwhile or necessary to extend the flags for this situation to neighboring pixels, given that the group 2 pixels are only barely saturated to begin with? Is there much of an effect on the neighbors?

Reply:
Yup.  See attached for an example; left is the B10.2 rate image, right is after this PR.  Most pixel values are about 1e-3 DN/s, the core of the hit is 493 DN/s, and the ring around it is anywhere from 3-100 DN/s (still orders of magnitude above the non-affected pixels).
<img width="1060" alt="Screenshot 2024-06-25 at 12 54 17 PM" src="https://github.com/spacetelescope/jwst/assets/11397442/9860cec1-43c6-4751-8db0-a56f1b907ac3">

Reply:
Wow, OK. So the answer is a definite "yes"!